packages/mock-block-dock/src/debug-view/dev-tools/datastore-graph-visualization.tsx

```diff
@@ -0,0 +1,368 @@
+            propertyTypeBaseUri,
+          }
+        : {
+            title: undefined,
+            propertyTypeBaseUri,
+          };
+    });
+
+  for (const option of options) {
+    const found = propertyTypes.find(({ title }) => title === option);
+
+    if (found) {
+      return getFallbackIfNotString(
+        entityToLabel.properties[found.propertyTypeBaseUri],
+      );
+    }
+  }
+
+  return getFallbackLabel();
+};
+
+type SeriesOption = GraphSeriesOption;
+
+// Combine an Option type with only required components and charts via ComposeOption
+type EChartOption = echarts.ComposeOption<SeriesOption>;
+
+const createDefaultEChartOptions = (params?: {
+  initialNodes?: GraphSeriesOption["nodes"];
+  initialEdges?: GraphSeriesOption["edges"];
+}): EChartOption => ({
+  color: ["#6F59EB"],
+  series: [
+    {
+      type: "graph",
+      layout: "force",
+      draggable: true,
+      /** @todo: find way to get this working, currently interferes with `draggable` */
+      // roam: true,
+      force: {
+        repulsion: 125,
+        edgeLength: 150,
+      },
+      label: {
+        fontSize: 16,
+      },
+      edgeLabel: {
+        fontSize: 10,
+      },
+      nodes: params?.initialNodes,
+      edges: params?.initialEdges,
+      // The size of the Node
+      symbolSize: 25,
+      edgeSymbol: ["none", "arrow"],
+    },
+  ],
+});
+
+// Register the required components
+echarts.use([GraphChart, SVGRenderer]);
+
+type EChartNode = {
+  id: string;
+  name: string;
+  fixed?: boolean;
+  x?: number;
+  y?: number;
+  label: {
+    show?: boolean;
+  };
+};
+
+const mapEntityToEChartNode = (
+  entity: Entity,
+  subgraph: Subgraph,
+): EChartNode => ({
+  id: JSON.stringify(entity.metadata.editionId),
+  name: parseLabelFromEntity(entity, subgraph),
+  label: { show: false },
+});
+
+type EChartEdge = {
+  id: string;
+  source: string;
+  target: string;
+  kind: string;
+  label: {
+    show?: boolean;
+  };
+};
+
+/** todo - render ontology-related edges */
+const mapGraphEdgeToEChartEdge = (
+  sourceEditionId: EntityEditionId,
+  targetEditionId: EntityEditionId,
+  edgeKind: OutwardEdge["kind"],
+): EChartEdge => ({
+  /** @todo - Can we do better than this, this assumes that this triple is unique, which it might not be */
+  id: `${JSON.stringify(sourceEditionId)}-${edgeKind}->${JSON.stringify(
+    targetEditionId,
+  )}`,
+  source: JSON.stringify(sourceEditionId),
+  target: JSON.stringify(targetEditionId),
+  kind: edgeKind,
+  label: { show: false },
+});
+
+const getSubgraphEntitiesAsEChartNodes = (subgraph: Subgraph): EChartNode[] => {
+  const allEntities = getEntities(subgraph);
+
+  /** @todo - Render link entities differently */
+  // const [linkEntities, nonLinkEntities] = partitionArrayByCondition(
+  //   allEntities,
+  //   (entity) =>
+  //     entity.linkData?.leftEntityId !== undefined &&
+  //     entity.linkData?.rightEntityId !== undefined,
+  // );
```

Comment:
I had an implementation which had split them up, but I didn't want to find all of the code around `EChartNode` and change if they have a border or whatever.

I do think we should play around with this a bit in the near future though for the sake of knowing how ergonomic the links are entities change even is.

Reply:
Would also be nice to have ontology nodes in the graph (potentially different colours, filterable, etc.)